conda_build_config.yaml

```diff
@@ -17,14 +18,19 @@ bzip2:
 cairo:
   - 1.16
 c_compiler:
+  # latest version is 12.2
```

Comment:
This comment - near the compiler choice is - inappropriate. Instead move the comment to `c_compiler_version:` or `cxx_compiler_version:`

Additionally, llvm/clang latest version is `15.0.2`.